ProjFS.Mac/PrjFSKextTests/HandleOperationTests.mm

```diff
@@ -706,5 +707,182 @@ - (void) testCloseWithoutModifed {
     XCTAssertFalse(MockCalls::DidCallFunction(ProviderMessaging_TrySendRequestAndWaitForResponse));
 }
 
+- (void) testDeleteWithNoVirtualizationRoot {
+    VirtualizationRoots_Cleanup();
+    testFileVnode->attrValues.va_flags = FileFlags_IsEmpty | FileFlags_IsInVirtualizationRoot;
+    HandleVnodeOperation(
+        nullptr,
+        nullptr,
+        KAUTH_VNODE_DELETE,
+        reinterpret_cast<uintptr_t>(context),
+        reinterpret_cast<uintptr_t>(testFileVnode.get()),
+        0,
+        0);
+
+    XCTAssertFalse(MockCalls::DidCallFunction(ProviderMessaging_TrySendRequestAndWaitForResponse));
+}
+
+- (void) testDeleteWithRequestResultFalse {
+    ProviderMessageMock_SetRequestResult(false);
+    testFileVnode->attrValues.va_flags = FileFlags_IsEmpty | FileFlags_IsInVirtualizationRoot;
+    HandleVnodeOperation(
+        nullptr,
+        nullptr,
+        KAUTH_VNODE_DELETE,
+        reinterpret_cast<uintptr_t>(context),
+        reinterpret_cast<uintptr_t>(testFileVnode.get()),
+        0,
+        0);
+
+   XCTAssertTrue(MockCalls::CallCount(ProviderMessaging_TrySendRequestAndWaitForResponse) == 1);
+   ProviderMessageMock_SetRequestResult(true);
```

Comment:
Could we update `(void) setUp` to always call:
- `ProviderMessageMock_SetRequestResult(true)`
- `ProviderMessageMock_SetSecondRequestResult(true)`
- `ProviderMessageMock_SetCleanupRootsAfterRequest(false)`

That way each test case won't have to worry about resetting their state.